src/Agent.Worker/Build/BuildCommandExtension.cs

```diff
@@ -16,6 +16,8 @@ public sealed class BuildCommandExtension : AgentService, IWorkerCommandExtensio
 
         public string CommandArea => "build";
 
+        public string SupportedHostType => string.Empty;
```

Comment:
A string doesn't make sense because it doesn't account for the scenario of only *some* host types supported. We have four host types today Build, Release, Deployment, and PoolMaintenance.

There should be a flags enum:

```csharp
[Flags]
public enum HostTypes
{
  None = 0,
  Build = 1,
  Deployment = 2,
  PoolMaintenance = 4,
  Release = 8,
  All = Build | Deployment | PoolMaintenance | Release,
}
```

The command interface should be:
```csharp
public HostTypes SupportedHostTypes { get; }
```

And variables class should expose a property
```csharp
public HostTypes System_HostType => GetEnum<HostTypes>(Constants.Variables.System.HostType) ?? HostTypes.None;
```

And IJobExtension interface should change:
```chsarp
public HostTypes HostType { get; }
```

And then worker command manager condition should change to:
```csharp
                if (!extension.SupportedHostTypes.HasFlag(context.Variables.HostType))
```